build.sh

```diff
@@ -275,28 +277,27 @@ find_architecture() {
 find_num_cores() {
     $ECHO "Finding num cores..."
     NUM_CORES=7ZZ
-    uname_s=`uname -s`
+    uname_s=$(uname -s)
     check_ret uname
     case $uname_s in
         CYGWIN_NT-5.2-WOW64 | *CYGWIN_NT* | *CYGWIN* | MINGW32*) NUM_CORES=$NUMBER_OF_PROCESSORS;;
         *darwin* | *Darwin* | *linux* | *Linux*) NUM_CORES=$(getconf _NPROCESSORS_ONLN);;
     esac
 }
 
-write_test_program() {
+echo_test_program() {
     #! Must be 'echo'
-    echo "#include <stdio.h>" > $C_WORD.c
-    echo "int main(){printf(\"%ld\", (long)(8*sizeof(void*))); return 0; }" >> $C_WORD.c
+    echo -e "int main(){ return (long)(8*sizeof(void*)); }"
 }
 
 c_find_word_size() {
     $ECHO "Finding WORD..."
-    C_WORD=factor-word-size
-    write_test_program
-    $CC -o $C_WORD $C_WORD.c
-    WORD=$(./$C_WORD)
-    check_ret $C_WORD
-    $DELETE -f $C_WORD*
+    C_WORD="factor-word-size"
```

Comment:
Most of the quotes you've added makes sense to me, but this (and one other I'll flag) seem unnecessary. Are you doing this purely for consistency? If so, why not also e.g. inside the conditionals on lines 411-412, for example?

(I'm not saying this wrong; I'm just genuinely unclear.)

Reply:
any references to disk filenames I personally think should be in quotes, mostly if these become variables and / or possibly contain spaces in the future. I know `cd factor` is perfectly legal (and i use it every day) but in a build script it feels like a bug waiting to happen.

Reply:
as for the conditionals at line 411, I know `$OS` etc will never contain spaces and putting `linux` in quotes despite `==` working on barewords does seem pointless.